### ISSUE TITLE
Clash of mapbox-java/MAS dependency

### ISSUE THREAD

**Issue by LukasPaczos** [mapbox/mapbox-gl-native#11417]
We have two sources of different versions of mapbox-java/MAS. One coming from new telemetry library (MAS `v2.2.6`) and other directly from Maps SDK (mapbox-java `v3.0.0-beta`).

When trying to build a release version of the app that depends on the Maps SDK `proguard` execution fails with warnings:
```
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find superclass or interface com.google.android.gms.common.api.GoogleApiClient$ConnectionCallbacks
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find superclass or interface com.google.android.gms.common.api.GoogleApiClient$OnConnectionFailedListener
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find superclass or interface com.google.android.gms.location.LocationListener
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient$ConnectionCallbacks
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient$OnConnectionFailedListener
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.LocationListener
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient$Builder
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient$Builder
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.LocationServices
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient$Builder
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.LocationServices
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.FusedLocationProviderApi
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.FusedLocationProviderApi
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.api.GoogleApiClient
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.common.ConnectionResult
Warning:com.mapbox.android.core.location.GoogleLocationEngine: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.GoogleLocationEngine$1$1: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.GoogleLocationEngine$1$2: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.GoogleLocationEngine$1$3: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.GoogleLocationEngine$1$4: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.utils.turf.TurfMeasurement
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.Point
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.utils.turf.TurfMeasurement
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.directions.v5.models.DirectionsRoute
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.directions.v5.models.RouteLeg
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.directions.v5.models.LegStep
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.directions.v5.models.DirectionsRoute
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.api.directions.v5.models.DirectionsRoute
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.geojson.LineString
Warning:com.mapbox.android.core.location.MockLocationEngine: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.MockLocationEngine$LocationUpdateRunnable: can't find referenced class com.mapbox.services.commons.models.Position
Warning:com.mapbox.android.core.location.UpdateGoogleRequestPriority: can't find referenced class com.google.android.gms.location.LocationRequest
Warning:com.mapbox.core.utils.ColorUtils: can't find referenced class java.awt.Color
Warning:com.mapbox.geojson.BoundingBox: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.Feature: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.FeatureCollection: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.GeometryCollection: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.LineString: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.MultiLineString: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.MultiPoint: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.MultiPolygon: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.Point: can't find referenced class com.google.auto.value.AutoValue
Warning:com.mapbox.geojson.Polygon: can't find referenced class com.google.auto.value.AutoValue
Warning:retrofit2.OkHttpCall: can't find referenced class javax.annotation.concurrent.GuardedBy
Warning:retrofit2.Platform$Java8: can't find referenced class org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
```

To resolve the issue we need to either remove MAS dependency from the telemetry library, or unify the versions, and bump telemetry library dependency within Maps SDK.

/cc @Guardiola31337 @electrostat

**Comment by electrostat** [comment]
Working through the Proguard issue here: https://github.com/mapbox/mapbox-events-android/pull/81

**Comment by tobrun** [comment]
Resolved with #11434 and https://github.com/mapbox/mapbox-gl-native/pull/11474
